python/pyspark/sql/tests/test_pandas_udf_grouped_map.py

```diff
@@ -29,6 +29,25 @@
     pandas_requirement_message, pyarrow_requirement_message
 from pyspark.testing.utils import QuietTest
 
+if have_pandas:
+    import pandas as pd
+    from pandas.util.testing import assert_frame_equal as pd_assert_frame_equal
+
+if have_pyarrow:
+    import pyarrow as pa
+
+
+def assert_frame_equal(left, right):
```

Comment:
This would fail if you don't have pandas with an error that it doesn't know what `pd_assert_frame_equal` is, since it's only imported above, conditional on pandas being imported already. This is only seen by this file and wouldn't change anything elsewhere.

I'm going to remove this though, and only define `check_column_type` here. That should make it clearer I think.